cmake/FindZSTD.cmake

```diff
@@ -23,6 +23,13 @@ set(ZSTD_STATIC_LIB_NAME
     ${CMAKE_STATIC_LIBRARY_PREFIX}zstd${ZSTD_STATIC_LIB_SUFFIX})
 
 # First, find via if specified ZTD_ROOT
+if (NOT ZSTD_ROOT)
+  # We look for an environment variable of the same name too, because
+  # that can be provided by a tool like nix-shell or docker. Wo do this
+  # so the user doesn't have to remember to add "-DZSTD_ROOT=..." to their
+  # cmake invocations.
```

Comment:
Having this for some variables but not all is really confusing and makes maintenance of the CMake way harder. I'm already always afraid to change CMake code because it might break the Nix build.

Reply:
This is essentially a workaround for a bug in the generated `.pc` files in ZSTD. We can probably get rid of `FindZSTD.cmake` when upgrading to the next Arrow release thanks to https://github.com/apache/arrow/pull/7388.